code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -262,62 +245,108 @@
 		client.time_died_as_mouse = world.time
 	..(gibbed)
 
+/mob/living/simple_animal/mouse/say_quote(text)
+	if(!text)
+		return "squeaks, \"...\"";	//not the best solution, but it will stop a large number of runtimes. The cause is somewhere in the Tcomms code
+	return "squeaks, [text]";
+
+/mob/living/simple_animal/mouse/singularity_act()
+	if(!(src.flags & INVULNERABLE))
+		investigation_log(I_SINGULO,"has been consumed by a singularity")
+		gib()
+		return 0
+
+/mob/living/simple_animal/mouse/bullet_act(var/obj/item/projectile/Proj)
+	..()
```

Comment:
You would actually not call ..() in this case if you wanted to keep the old behavior, but I actually think this may be better because now that I think of it, if you shoot a rocket at a mouse it might not explode since it will exit before the on_hit. This fixes that.

Reply:
Returning the value from supercall `. = ..()` might also be relevant but I'm not sure.

Reply:
I actually didn't think about processing the on_hit behavior when doing this so that was just a lucky coincidence. This preserves the original behavior of regular mice splatting to stun projectiles while mice with an unusually high max HP will process projectile damage normally.

Although I'm pretty sure it will runtime if there's no projectile so I should probably fix that.